src/App.js

```diff
@@ -92,11 +92,21 @@ function App() {
         .get()
         .then((doc) => {
           if (doc.exists) {
+            // console.log(Object.keys(doc.data()))
             const tt = {};
+
             tt.identifiedCourses = doc.data().identifiedCourses;
             tt.identifiedSegments = doc.data().identifiedSegments;
             tt.identifiedSlots = doc.data().identifiedSlots;
 
+            if (
+              Object.keys(doc.data()).indexOf('identifiedCourseNames') !== -1
```

Comment:
This could use something simpler:

Something like `if('identifiedCourseNames' in doc.data())` or use hasOwnProperty. 

You can also use [optionalChaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) but then your sentinel will be `undefined` instead of `null`.